Intersect.Client.Core/Interface/Game/TargetContextMenu.cs

```diff
@@ -0,0 +1,253 @@
+    private readonly MenuItem _friendMenuItem;
+    private readonly MenuItem _guildMenuItem;
+    private readonly MenuItem _privateMessageMenuItem;
+    private readonly Player? _me;
+    private Entity? _entity;
+
+    public TargetContextMenu(Canvas gameCanvas) : base(gameCanvas, nameof(TargetContextMenu))
+    {
+        IsHidden = true;
+        IconMarginDisabled = true;
+        Children.Clear();
+
+        _me = Globals.Me;
+
+        _tradeMenuItem = AddItem(Strings.EntityContextMenu.Trade);
+        _tradeMenuItem.Clicked += tradeRequest_Clicked;
+
+        _partyMenuItem = AddItem(Strings.EntityContextMenu.Party);
+        _partyMenuItem.Clicked += invite_Clicked;
+
+        _friendMenuItem = AddItem(Strings.EntityContextMenu.Friend);
+        _friendMenuItem.Clicked += friendRequest_Clicked;
+
+        _guildMenuItem = AddItem(Strings.EntityContextMenu.Guild);
+        _guildMenuItem.Clicked += guildRequest_Clicked;
+
+        _privateMessageMenuItem = AddItem(Strings.EntityContextMenu.PrivateMessage);
+        _privateMessageMenuItem.Clicked += privateMessageRequest_Clicked;
+
+        LoadJsonUi(GameContentManager.UI.InGame, Graphics.Renderer?.GetResolutionString());
+    }
+
+    public void ToggleHidden(object? target)
+    {
+        if (target == null || _me == null)
+        {
+            return;
+        }
+
+        float posX, posY, newX, newY;
+
+        switch (target)
+        {
+            case Button button:
+            {
+                _entity = _me.TargetBox?.MyEntity;
+                posX = button.LocalPosToCanvas(new Point(0, 0)).X;
+                posY = button.LocalPosToCanvas(new Point(0, 0)).Y;
+                newX = posX;
+                newY = posY + button.Height;
+
+                if (newX + Width >= Canvas?.Width)
+                {
+                    newX = posX - Width + button.Width;
+                }
+
+                if (newY + Height >= Canvas?.Height)
+                {
+                    newY = posY - Height;
+                }
+
+                break;
+            }
+            case Entity en:
+            {
+                if (en is not Player || en == _me)
+                {
+                    return;
+                }
+
+                var mousePos = Graphics.ConvertToWorldPoint(Globals.InputManager.MousePosition);
+                bool isHovered = en.WorldPos.Contains(mousePos.X, mousePos.Y);
+
+                if (!isHovered)
+                {
+                    return;
+                }
+
+                _entity = en;
+                posX = InputHandler.MousePosition.X;
+                posY = InputHandler.MousePosition.Y;
+                newX = posX;
+                newY = posY;
+
+                if (newX + Width >= Canvas?.Width)
+                {
+                    newX = posX - Width;
+                }
+
+                if (newY + Height >= Canvas?.Height)
+                {
+                    newY = posY - Height;
+                }
```

Comment:
```
if (Canvas is {} canvas)
{
    ...
}
```

less than/greater than comparisons with nullables is sketchy, not clear what the result at a glance